libraries/rush-lib/src/logic/DependencySpecifier.ts

```diff
@@ -92,6 +92,7 @@ export class DependencySpecifier {
     if (versionSpecifier.startsWith('workspace:')) {
       this.specifierType = DependencySpecifierType.Workspace;
       this.versionSpecifier = versionSpecifier.slice(this.specifierType.length + 1).trim();
+      // "workspace:some-package@^1.2.3" should be resolved as alias
```

Comment:
This kind of check is already being performed for `npm:` aliases by **ApprovedPackagesChecker.ts**:

https://github.com/microsoft/rushstack/blob/c42ab49e21431dc036c3ddab4090ef772d480415/libraries/rush-lib/src/logic/ApprovedPackagesChecker.ts#L68-L79

But the above logic is only used for the `browserApprovedPackages` feature. The `rush build` dependency graph instead seems to be based on this API:

https://github.com/microsoft/rushstack/blob/c42ab49e21431dc036c3ddab4090ef772d480415/libraries/rush-lib/src/api/RushConfigurationProject.ts#L395-L402

For both cases, we just need to consider `dependencySpecifier.aliasTarget.packageName` rather than assuming that the `"dependencies"` key is the project name.

@dmichon-msft

Reply:
Add `class WorkspaceSpec` to parse workspace protocol.
This idea came from [pnpm](https://github.com/pnpm/pnpm/blob/27c33f0319f86c45c1645d064cd9c28aada80780/workspace/pkgs-graph/src/index.ts#L49)
I'm not sure whether the `rush workspace protocol` is equal to `pnpm workspace protocol`. So it's copied instead of importing `@pnpm/npm-resolver`